From f8661f2c5543668fd1dcaf8a8bb76ca59082024e Mon Sep 17 00:00:00 2001
From: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>
Date: Thu, 11 Oct 2018 08:49:29 -0700
Subject: [PATCH 04/15] h264e: remove redundunt
 MfxHwH264Encode::ReadFrameData()

The function is not used in the project.

Signed-off-by: Dmitry Rogozhkin <dmitry.v.rogozhkin@intel.com>
---
 .../shared/include/mfx_h264_enc_common_hw.h   |  9 ----
 .../shared/src/mfx_h264_enc_common_hw.cpp     | 50 -------------------
 2 files changed, 59 deletions(-)

diff --git a/_studio/mfx_lib/shared/include/mfx_h264_enc_common_hw.h b/_studio/mfx_lib/shared/include/mfx_h264_enc_common_hw.h
index f4afb86..8c620a0 100644
--- a/_studio/mfx_lib/shared/include/mfx_h264_enc_common_hw.h
+++ b/_studio/mfx_lib/shared/include/mfx_h264_enc_common_hw.h
@@ -29,7 +29,6 @@
 #include <vector>
 #include <assert.h>
 #include <memory>
-#include "vm_file.h"
 #include "mfx_ext_buffers.h"
 #include "mfxfei.h"
 #include "mfxbrc.h"
@@ -824,14 +823,6 @@ namespace MfxHwH264Encode
         mfxFrameData const & src,
         mfxFrameInfo const & info);
 
-
-    mfxStatus ReadFrameData(
-        vm_file *            file,
-        mfxU32               frameNum,
-        VideoCORE *          core,
-        mfxFrameData const & data,
-        mfxFrameInfo const & info);
-
     mfxExtBuffer * GetExtBuffer(
         mfxExtBuffer ** extBuf,
         mfxU32          numExtBuf,
diff --git a/_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp b/_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp
index 0be4cbf..87cf433 100644
--- a/_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp
+++ b/_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp
@@ -6905,56 +6905,6 @@ mfxStatus MfxHwH264Encode::CopyFrameDataBothFields(
     return core->DoFastCopyWrapper(&surfDst,MFX_MEMTYPE_INTERNAL_FRAME|MFX_MEMTYPE_DXVA2_DECODER_TARGET|MFX_MEMTYPE_FROM_ENCODE, &surfSrc, MFX_MEMTYPE_EXTERNAL_FRAME|MFX_MEMTYPE_SYSTEM_MEMORY);
 }
 
-
-
-mfxStatus MfxHwH264Encode::ReadFrameData(
-    vm_file *            file,
-    mfxU32               frameNum,
-    VideoCORE *          core,
-    mfxFrameData const & fdata,
-    mfxFrameInfo const & info)
-{
-    mfxFrameData data = fdata;
-    FrameLocker lock(core, data, false);
-
-    if (file != 0 && data.Y != 0 && data.UV != 0)
-    {
-        mfxU32 frameSize = (info.Height * info.Width * 5) / 4;
-        if (vm_file_fseek(file, frameSize * frameNum, VM_FILE_SEEK_SET) != 0)
-        {
-            MFX_RETURN(MFX_ERR_NOT_FOUND);
-        }
-
-        for (mfxU32 i = 0; i < info.Height; i++)
-        {
-            if(!vm_file_fread(data.Y + i * data.Pitch, 1, info.Width, file))
-            {
-                return MFX_ERR_UNKNOWN;
-            }
-        }
-
-        for (mfxI32 y = 0; y < info.Height / 2; y++)
-            for (mfxI32 x = 0; x < info.Width; x += 2)
-            {
-                if(!vm_file_fread(data.UV + y * data.Pitch + x, 1, 1, file))
-                {
-                    return MFX_ERR_UNKNOWN;
-                }
-            }
-
-        for (mfxI32 y = 0; y < info.Height / 2; y++)
-            for (mfxI32 x = 1; x < info.Width; x += 2)
-            {
-                if(!vm_file_fread(data.UV + y * data.Pitch + x, 1, 1, file))
-                {
-                    return MFX_ERR_UNKNOWN;
-    }
-            }
-    }
-
-    return MFX_ERR_NONE;
-}
-
 mfxExtBuffer* MfxHwH264Encode::GetExtBuffer(mfxExtBuffer** extBuf, mfxU32 numExtBuf, mfxU32 id, mfxU32 offset)
 {
     if (extBuf != 0)
-- 
2.17.2

